future-apps/actor/test/actor.js

```diff
@@ -0,0 +1,351 @@
+const Actor = artifacts.require('Actor')
+
+const { assertRevert, assertInvalidOpcode } = require('@aragon/test-helpers/assertThrow')
+const { hash: namehash } = require('eth-ens-namehash')
+const ethutil = require('ethereumjs-util')
+const getBalance = require('@aragon/test-helpers/balance')(web3)
+const web3Call = require('@aragon/test-helpers/call')(web3)
+const web3Sign = require('@aragon/test-helpers/sign')(web3)
+const { encodeCallScript, EMPTY_SCRIPT } = require('@aragon/test-helpers/evmScript')
+const assertEvent = require('@aragon/test-helpers/assertEvent')
+const ethABI = new (require('web3-eth-abi').AbiCoder)()
+const getEvent = (receipt, event, arg) => { return receipt.logs.filter(l => l.event == event)[0].args[arg] }
+
+const ACL = artifacts.require('ACL')
+const AppProxyUpgradeable = artifacts.require('AppProxyUpgradeable')
+const EVMScriptRegistryFactory = artifacts.require('EVMScriptRegistryFactory')
+const DAOFactory = artifacts.require('DAOFactory')
+const Kernel = artifacts.require('Kernel')
+const KernelProxy = artifacts.require('KernelProxy')
+
+const EtherTokenConstantMock = artifacts.require('EtherTokenConstantMock')
+const KernelDepositableMock = artifacts.require('KernelDepositableMock')
+
+const ExecutionTarget = artifacts.require('ExecutionTarget')
+const DesignatedSigner = artifacts.require('DesignatedSigner')
+
+const NULL_ADDRESS = '0x00'
+
+const EIP165_SUPPORT_INTERFACE_ID = '0x01ffc9a7'
+const EIP165_SUPPORT_INVALID_ID = '0xffffffff'
+
+contract('Actor app', (accounts) => {
+  let daoFact, actorBase, acl, actor, actorId
+
+  let ETH, ANY_ENTITY, APP_MANAGER_ROLE, EXECUTE_ROLE, RUN_SCRIPT_ROLE, PRESIGN_HASH_ROLE, DESIGNATE_SIGNER_ROLE, ISVALIDSIG_INTERFACE_ID
+
+  const root = accounts[0]
+
+  const encodeFunctionCall = (contract, functionName, ...params) =>
+    contract[functionName].request(...params).params[0]
```

Comment:
👀Just curious, where did you find this API? Is it documented anywhere?

Reply:
I discovered this API when working on Aragon 0.1's BylawsLib and I have been using it ever since lol.

Reply:
Pretty sure it is an internal thing we are not supposed to be using